planner/mapper/mapper.go

```diff
@@ -844,6 +845,11 @@ func resolveInnerFilterDependencies(
 			// If the key index is outside the bounds of the child mapping array, then
 			// this is not a relation/join and we can add it to the fields and
 			// continue (no child props to process)
+			for _, field := range existingFields {
```

Comment:
question: You are removing duplicate fields that didn't impact query results at the expense of an extra for loop for every field. Could you explain why you had to do this? Or guide me towards the explanation if it exists already.

Reply:
The linked issue does cover this (#1615), but it does require some imagination in order to connect the dots.

Reply:
It doesn't. It show state of variables. I understand that there was duplicates and doing this removes the duplicates. I'd like to know if it's just for cleanliness of if it has a direct impact on something else which could help support the extra for loop.

Reply:
I already replied to Andy on this matter. Here is my quote:
> So far I didn't see any public impact. I'm not sure why. The collected fields are used internally and I assume that so far the code didn't rely on uniqueness of the fields in the list.
I noticed this by observing weird behaviour while working on indexes.